src/main/clojure/clara/rules/engine.cljc

```diff
@@ -1729,7 +1729,25 @@
                                :listener listener}]
 
     (loop [next-group (mem/next-activation-group transient-memory)
-           last-group nil]
+           last-group nil
+           flushed-updates-count 0]
+
+      (when (> flushed-updates-count max-cycles)
+        (throw (ex-info
+                (str "It appears that the rules are in an infinite loop."
+                     " Incorrect use of truth maintenance is a frequent source of such loops; see the website."
+                     " If the rules are not in fact in a loop it is possible to increase the ceiling by use of the :max-cycles setting."
+                     " See the clara.rules/fire-rules docstring for details.")
+                {:clara-rules/infinite-loop-suspected true
+                 :listeners (try
+                              (let [p-listener (l/to-persistent! listener)]
+                                (if (l/null-listener? p-listener)
+                                  []
+                                  (l/get-children p-listener)))
+                              (catch #?(:clj Exception :cljs :default)
+                                  listener-exception
```

Comment:
Indentation makes this confusing.

Reply:
How would you indent it?  This looks fairly normal to me but open to suggestions.

Reply:
```
(catch #?(:clj Exception :cljs :default)  listener-exception
     listener-exception)
```
is probably how i would write it, but that just because i am used to seeing:
```
(try 
  <some logic>
  (catch Exception ex
    <do something>))
```
though the reader conditional makes this a bit different.

Reply:
I think it was weird to see the first `listener-exception` on the next line. It just makes it blend with the body.
I don't consider it to be a critical point though if others thing it is fine how it is. I know reader-conditionals can make things get a bit weird.